www/js/diary/list.js

```diff
@@ -1,17 +1,23 @@
+'use strict';
+
 angular.module('emission.main.diary.list',['ui-leaflet',
                                       'ionic-datepicker',
-                                      'emission.main.common.services',
                                       'emission.incident.posttrip.manual',
                                       'emission.services',
-                                      'ng-walkthrough', 'nzTour', 'angularLocalStorage'])
+                                      'ng-walkthrough', 'nzTour', 'angularLocalStorage', 'emission.splash.startprefs',
+                                        'emission.splash.updatecheck',
+                                        'emission.main.metrics.factory',
+                                        'emission.stats.clientstats',
+                                        'angularLocalStorage'])
 
 .controller("DiaryListCtrl", function($window, $scope, $rootScope, $ionicPlatform, $state,
                                     $ionicScrollDelegate, $ionicPopup,
                                     $ionicLoading,
-                                    $ionicActionSheet,
+                                    $ionicActionSheet, $location, storage,
```

Comment:
where are you using the `$location` service?
Only instance that I can find is commented out.
Please remove commented out code and this service from the dependencies.